cmd/sonarExecuteScan_test.go

```diff
@@ -0,0 +1,159 @@
+package cmd
+
+import (
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+
+	piperHttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/mock"
+	FileUtils "github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockDownloader struct {
+	shouldFail    bool
+	requestedURL  string
+	requestedFile string
+}
+
+func (m *mockDownloader) DownloadFile(url, filename string, header http.Header, cookies []*http.Cookie) error {
+	m.requestedURL = url
+	m.requestedFile = filename
+	if m.shouldFail {
+		return errors.New("something happened")
+	}
+	return nil
+}
+
+func (m *mockDownloader) SetOptions(options piperHttp.ClientOptions) {
+	return
+}
+
+func mockFileUtilsExists(filename string) (bool, error) {
+	return true, nil
+}
+
+func mockExecLookPath(executable string) (string, error) {
+	if executable == "local-sonar-scanner" {
+		return "/usr/bin/sonar-scanner", nil
+	}
+	return "", errors.New("something happened")
+}
+
+func mockFileUtilsUnzip(t *testing.T, expectSrc string) func(string, string) ([]string, error) {
+	return func(src, dest string) ([]string, error) {
+		assert.Equal(t, path.Join(dest, expectSrc), src)
+		return []string{}, nil
+	}
+}
+
+func mockOsRename(t *testing.T, expectOld, expectNew string) func(string, string) error {
+	return func(old, new string) error {
+		assert.Regexp(t, expectOld, old)
+		assert.Equal(t, expectNew, new)
+		return nil
+	}
+}
+
+func mockOsRemove(t *testing.T) func(string) error {
+	return func(filename string) error {
+		//TODO: tempDir name not known here
+		//assert.Equal(t, "", filename)
+		return nil
+	}
+}
+
+func TestSonarHandlePullRequest(t *testing.T) {
```

Comment:
what about testing `runSonar`, e.g. success & error case?

I guess we should increase coverage a bit ...